.github/workflows/development-environment.yml

```diff
@@ -0,0 +1,83 @@
+name: development environment
+on:
+  pull_request:
+    paths:
+    - 'Brewfile'
+    - 'Makefile'
+    - '.github/workflows/development-environment.yml'
+    - 'requirements-*.txt'
+    - 'scripts/*'
+    - 'src/sentry/runner/commands/devserver.py'
+    - 'src/sentry/runner/commands/devservices.py'
+
+jobs:
+  dev-environment:
+    name: python set up
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        os: [ macos-11.0 ]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install prerequisites
+        if: ${{ runner.os == 'macOS' }}
+        # Depending on the host watchman will not install (read: intermittent issue),
+        # thus, we mark this as continue-on-error
+        continue-on-error: true
+        # Xcode CLI & brew are already installed, thus, no need to call xcode-select install
+        # XXX: Can the Brew set up be cached?
+        run: |
+          brew bundle
+
+      - name: Set environment variables
+        shell: bash
+        run: |
+          echo "PYENV_ROOT=$HOME/.pyenv" >> $GITHUB_ENV
+          echo "VOLTA_HOME=$HOME/.volta" >> $GITHUB_ENV
+          echo "PATH=$HOME/.pyenv/shims:$HOME/.volta/bin:$PATH" >> $GITHUB_ENV
+
+      - name: Install Python (via Pyenv)
+        id: python-version
+        shell: bash
+        # XXX: Locally, pyenv makes 3.6.10 be the selected version in use,
+        # however, in here it selects /usr/local/bin/python. I'm using `pyenv global` to work around it
+        # XXX: Can we cache the installed Python environments?
+        run: |
+          make setup-pyenv
+          pyenv global 3.6.10
+          echo "::set-output name=python-version::$(python -V  | sed s/Python\ //g)"
+
+      - name: Setup pip
+        uses: ./.github/actions/setup-pip
+        id: pip
+
+      - name: Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ steps.pip.outputs.pip-cache-dir }}
+            ~/.pyenv
+          key: ${{ runner.os }}-pip-py${{ steps.python-version.outputs.python-version }}-${{ hashFiles('**/requirements-*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-py${{ steps.python-version.outputs.python-version }}
+
+      - name: Set up development environment (as per docs)
+        run: |
+          python -m pip install virtualenv
```

Comment:
Once we drop 2.7 once and for all, being on a python >= 3.4 means we can rely on the builtin `venv` instead of external `virtualenv`. I'd like to remove `virtualenv` across the board for devenv, it'll just be an extra moving part that isn't needed.

Reply:
I agree with you. I tried it a few times. It works without any difference as far as I can tell.